packages/in_app_purchase/in_app_purchase_storekit/lib/src/in_app_purchase_storekit_platform.dart

```diff
@@ -355,6 +355,34 @@ class InAppPurchaseStoreKitPlatform extends InAppPurchasePlatform {
     return _useStoreKit2;
   }
 
+  /// Checks if the user is eligible for a introductory offer (StoreKit2 only).
+  ///
+  /// Throws [PlatformException] if StoreKit2 is not enabled, if the product is not found,
+  /// if the product is not a subscription, or if any error occurs during the eligibility check.
+  ///
+  /// [PlatformException.code] can be one of:
+  /// - `storekit2_not_enabled`
+  /// - `storekit2_unsupported_platform_version`
+  /// - `storekit2_failed_to_fetch_product`
+  /// - `storekit2_not_subscription`
+  /// - `storekit2_eligibility_check_failed`
+  Future<bool> isIntroductoryOfferEligible(
```

Comment:
What happens if this is called on an older version of iOS that doesn't have this API?

Reply:
@LongCatIsLooong Could you suggest, what to do in this case?